Library/Homebrew/cmd/audit.rb

```diff
@@ -827,12 +827,26 @@ def audit_version
   def audit_checksum
     return unless checksum
 
-    case checksum.hash_type
-    when :md5
-      problem "MD5 checksums are deprecated, please use SHA1 or SHA256"
-      return
-    when :sha1   then len = 40
-    when :sha256 then len = 64
+    strict = ARGV.include? "--strict"
+
+    if strict
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
+        return
+      when :sha1
+        problem "SHA1 checksums are deprecated, please use SHA256"
+        return
+      when :sha256 then len = 64
+      end
+    else
+      case checksum.hash_type
+      when :md5
+        problem "MD5 checksums are deprecated, please use SHA256"
```

Comment:
Agreed. The ` case ` is somewhat limiting, It wouldn't let me stick the ` if strict ` statement inside of it without my Ruby auditor freaking out locally.

> perhaps turn the case statement into an if one.

As in ` if checksum.hash_type `?